frame/scheduler/src/lib.rs

```diff
@@ -601,6 +666,17 @@ impl<T: Trait> schedule::Named<T::BlockNumber, <T as Trait>::Call, T::PalletsOri
 	fn cancel_named(id: Vec<u8>) -> Result<(), ()> {
 		Self::do_cancel_named(None, id).map_err(|_| ())
 	}
+
+	fn reschedule_named(
+		id: Vec<u8>,
+		when: DispatchTime<T::BlockNumber>,
+	) -> Result<Self::Address, DispatchError> {
+		Self::do_reschedule_named(id, when)
+	}
+
+	fn next_dispatch_time(id: Vec<u8>) -> Result<T::BlockNumber, ()> {
+		Lookup::<T>::get(id).and_then(|(when, index)| Agenda::<T>::get(when).get(index as usize).map(|_| when)).ok_or(())
```

Comment:
would something like this work?

```suggestion
		Lookup::<T>::get(id).and_then(Self::next_dispatch_time)
```